Split storage

Ying-metaswitch opened the pull request:
Changes to HEAT templates to use split storage architecture:
1. Security groups for the new nodes, often split into inbound traffic group and outbound traffic group
2. Public hostname and DNS config 
3. Install new node packages (need the package repo to be public though)

Test done:
1. Deployment using HEAT, for various cluster size (Sprout, Dime, Vellum and Homer, since these are the nodes being changed).
2. Monit summary shows the relevant process running.
3. Live test suite was run for different cluster size, fails in one case - ISC B2BUA, not caused by this change though
4. Live call: provision in Ellis, register in Blink with Bono IP, made calls past 30 secs
NB. Succeed only after manually adding Bono IP (created per deployment) to bind file in internal DNS server.

Ying-metaswitch committed:
Split storage major implementations for security groups.

Ying-metaswitch committed:
Split storage major implementations for clearwater yaml.

Ying-metaswitch committed:
Split storage major implementations for sprout yaml.

Ying-metaswitch committed:
Split storage major implementations for dime yaml.

Ying-metaswitch committed:
Split storage major implementations for vellum yaml.

Ying-metaswitch committed:
Split storage major implementations for vellum yaml.

Ying-metaswitch committed:
Split storage major implementations for ellis shared config.

johadalin commented on ellis.yaml:
```diff
@@ -189,13 +189,20 @@ resources:
             # Deployment definitions
             home_domain=__zone__
             sprout_hostname=sprout.__zone__
+            sprout_hostname_mgmt=sprout.__zone__:9886
```
Do you know what this is used for? 
I don't think you have opened traffic to this port in the security group changes/

johadalin commented on ellis.yaml:
```diff
@@ -189,13 +189,20 @@ resources:
             sprout_hostname=sprout.__zone__
+            sprout_hostname_mgmt=sprout.__zone__:9886
             hs_hostname=hs.__zone__:8888
+            hs_hostname_mgmt=hs.__zone__:8886
```
Same as above

johadalin commented on ellis.yaml:
```diff
@@ -189,13 +189,20 @@ resources:
-            sprout_registration_store=sprout.__zone__
-            ralf_session_store=ralf.__zone__
-
+            cdf_identity=cdf.__zone__.com
```
I think you've taken these direct from the Chef setup. This is mostly fine, but i don't think all of it makes sense. For example, we don't set up a CDF as part of the Heat deployment, and so don't set up a cdf.__Zone__ DNS entry. I don't think we need to as part of this, so having it in shared config is a bit misleading.

We can have a chat about what we need here later

johadalin commented on security-groups.yaml:
```diff
@@ -291,54 +273,90 @@ resources:
-          #remote_group_id: { get_resource: ralf_sig } # omit remote_group_id to reference yourself
-        # Chronos timer pops
+          remote_group_id: { get_resource: dime_sig }
+        # Homestead
```
This, and the rule below, are for contacting the Cassandra process using Thrift. I think the other comments refer to the process being contacted, not where the traffic is coming from (Homestead, Homer); can these do the same?

johadalin commented on sprout.yaml:
```diff
@@ -227,25 +227,6 @@ resources:
-            EOF
-
             # Now install the software.
             DEBIAN_FRONTEND=noninteractive apt-get install sprout-node --yes --force-yes
```
We haven't discussed what we'll be doing about the sprout vs sprout-node package here. I don't know if we've agreed a longer term position here, but atm this will install more packages than we want

Ying-metaswitch replied:
Richard mentioned that in the split-storage architecture, the sprout-node will be updated to contain only Sprout, so installing node package for all nodes will work fine.

Ying-metaswitch committed:
Fixed security group name typos to successfully deploy.

johadalin reviewed (changes requested):
I've reviewed as far as commit b636337. A lot of the changes are good, following the design. I've put a number of comments throughout that i think we should go through at some point today.
Some high level feedback so far:

* A little bit more/clearer information on what is contained in both the pull request, and in each commit, would be useful while reviewing. I think it would be worth reading over some historical pull requests, and following the github updates for a while to get a feel for how we tend to do this on the team.
* I like that you've split the changes out into logical chunks. This helps clarify it a lot. The commit messages could be a bit more descriptive about what the actual changes are, and the first of the vellum commits should probably state that it's just copying the sprout file across (otherwise it's hard to know if it should be reviewed in detail or not).
* We should talk about testing this morning, and how it should fit into the PR flow

Ying-metaswitch committed:
Removed irrelevant shared config,changed security group Cassandra comment,changed Sprout Homer package name.

Ying-metaswitch committed:
Added Chronos config to vellum.

Ying-metaswitch commented:
New commits fixed some security group typos using error logs throw during deployment, removed some irrelevant shared config entries, and added Chronos config to Vellum. It also adds DNS server address to Chronos config.

johadalin commented on vellum.yaml:
```diff
@@ -178,7 +178,7 @@ resources:
 
             # Log all output to file.
-            exec > >(tee -a /var/log/clearwater-heat-homestead.log) 2>&1
+            exec > >(tee -a /var/log/clearwater-heat-sprout.log) 2>&1
```
Think you missed this. Should be pointing to a vellum log?

johadalin replied:
We should also update the log name for dime, at https://github.com/Metaswitch/clearwater-heat/pull/40/files#diff-fbfa58039a87d980a307ef70da669b18R181

johadalin commented on vellum.yaml:
```diff
@@ -265,10 +283,10 @@ resources:
     value: { get_attr: [ mgmt_floating_ip, floating_ip_address ] }
   private_mgmt_ip:
-    description: IP address in private management network
+    description: IP address in private signaling network
```
I think the changes here have got a bit confused. Don't think these entries needed changing

johadalin reviewed (commented):
Just a couple of comments for minor things, otherwise all looks good :+1: 

Can you also stick a brief summary of the different test scenarios you've run through, as a record? :)

Ying-metaswitch committed:
Final markup. Typo in log file name.
